src/MongoDB.Bson/Serialization/Serializers/ObjectSerializer.cs

```diff
@@ -92,7 +92,23 @@ public override object Deserialize(BsonDeserializationContext context, BsonDeser
                     var subType = binaryData.SubType;
                     if (subType == BsonBinarySubType.UuidStandard || subType == BsonBinarySubType.UuidLegacy)
                     {
-                        return binaryData.ToGuid();
+#pragma warning disable 618
+                        if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
+                        {
+                            return binaryData.ToGuid();
+                        }
+                        else
+                        {
+                            if (subType == BsonBinarySubType.UuidStandard)
+                            {
+                                return binaryData.ToGuid(GuidRepresentation.Standard);
+                            }
+                            else
+                            {
+                                return binaryData;
```

Comment:
This is a behavioral change (return a `BsonBinaryData` instance instead of a `Guid` instance).

I thought about throwing an exception here, but since there is no way to configure your way out of such an exception I decided not to.

Reply:
Just to hightlight that it's still under discussion